src/Blocks/components/conditional-tags/components/conditional-tags-options.js

```diff
@@ -89,217 +64,241 @@ export const ConditionalTagsOptions = (attributes) => {
-				icon={icons.conditionalVisibility}
-				label={__('Conditional visibility', 'eightshift-frontend-libs')}
-				checked={conditionalTagsUse}
-				onChange={(value) => {
-					setAttributes({ [getAttrKey('conditionalTagsUse', attributes, manifest)]: value });
-
-					if (!value) {
-						setAttributes({ [getAttrKey('conditionalTagsAction', attributes, manifest)]: undefined });
-						setAttributes({ [getAttrKey('conditionalTagsLogic', attributes, manifest)]: undefined });
-						setAttributes({ [getAttrKey('conditionalTagsRules', attributes, manifest)]: undefined });
-					} else {
-						setAttributes({ [getAttrKey('conditionalTagsRules', attributes, manifest)]: [] });
-					}
-				}}
-				noBottomSpacing={!conditionalTagsUse}
-				additionalClasses='es-font-weight-500'
-			/>
-
-			<Section showIf={conditionalTagsUse} noBottomSpacing>
+		<>
+		{formFields?.length < 1 ?
+			<PanelBody>
 				<Control
-					icon={icons.conditionH}
-					label={__('Rules', 'eightshift-forms')}
-					// Translators: %d refers to the number of active rules
-					subtitle={conditionalTagsRules?.length > 0 && sprintf(__('%d added', 'eightshift-forms'), conditionalTagsRules.length)}
+					icon={icons.conditionalVisibility}
+					label={__('Conditional visibility', 'eightshift-forms')}
+					additionalLabelClasses='es-font-weight-500'
 					noBottomSpacing
-					inlineLabel
 				>
-					<Button
-						variant='tertiary'
-						onClick={() => setIsModalOpen(true)}
-						className='es-rounded-1.5 es-w-9 es-h-center es-font-weight-500'
-					>
-						{conditionalTagsRules?.length > 0 ? __('Edit', 'eightshift-forms') : __('Add', 'eightshift-forms')}
-					</Button>
+					<IconLabel
+						icon={icons.warningFillTransparent}
+						label={__('Feature unavailable', 'eightshift-forms')}
+						subtitle={__('No fields have a name set', 'eightshift-forms')}
+						additionalClasses='es-nested-color-yellow-500!'
+						addSubtitleGap
+						standalone
+					/>
 				</Control>
+			</PanelBody> :
+			<PanelBody>
+				<IconToggle
+					icon={icons.conditionalVisibility}
+					label={__('Conditional visibility', 'eightshift-forms')}
+					checked={conditionalTagsUse}
+					onChange={(value) => {
+						setAttributes({ [getAttrKey('conditionalTagsUse', attributes, manifest)]: value });
 
-				{isModalOpen &&
-					<Modal
-						overlayClassName='es-conditional-tags-modal es-geolocation-modal'
-						className='es-modal-max-width-xxl es-rounded-3!'
-						title={<IconLabel icon={icons.conditionalVisibility} label={__('Conditional visibility', 'eightshift-forms')} standalone />}
-						onRequestClose={() => {
-							setIsModalOpen(false);
-							setIsNewRuleAdded(false);
-						}}
-						isDismissible={false}
+						if (!value) {
+							setAttributes({ [getAttrKey('conditionalTagsRules', attributes, manifest)]: undefined });
+						} else {
+							setAttributes({ [getAttrKey('conditionalTagsRules', attributes, manifest)]: [CONDITIONAL_TAGS_ACTIONS.HIDE, []]});
+						}
```

Comment:
could add a return instead of else
or just one ternary

```suggestion
							setAttributes({ [getAttrKey('conditionalTagsRules', attributes, manifest)]: !value ? undefined : [CONDITIONAL_TAGS_ACTIONS.HIDE, []]});
```